packages/venia-ui/lib/components/SearchBar/autocomplete.js

```diff
@@ -1,51 +1,33 @@
-import React, { useEffect } from 'react';
+import React from 'react';
 import { bool, func, shape, string } from 'prop-types';
-import { useFieldState } from 'informed';
-import { useQuery } from '@magento/peregrine';
+import { useAutocomplete } from '@magento/peregrine/lib/talons/SearchBar';
 
 import { mergeClasses } from '../../classify';
 import PRODUCT_SEARCH from '../../queries/productSearch.graphql';
 import Suggestions from './suggestions';
 import defaultClasses from './autocomplete.css';
 
+const MESSAGES = new Map()
+    .set('ERROR', 'An error occurred while fetching results.')
+    .set('LOADING', 'Fetching results...')
+    .set('PROMPT', 'Search for a product')
+    .set('EMPTY_RESULT', 'No results were found.')
+    .set('RESULT_SUMMARY', (_, resultCount) => `${resultCount} items`);
+
 const Autocomplete = props => {
     const { setVisible, visible } = props;
-
-    const [queryResult, queryApi] = useQuery(PRODUCT_SEARCH);
-    const { data, error, loading } = queryResult;
-    const { resetState, runQuery, setLoading } = queryApi;
-
-    const { value } = useFieldState('search_query');
-    const valid = value && value.length > 2;
+    const talonProps = useAutocomplete({ query: PRODUCT_SEARCH, visible });
+    const { messageType, resultCount, queryResult, value } = talonProps;
+    const { data } = queryResult;
```

Comment:
I was thinking about this -- do we want to return the `queryResult` or do we want to normalize within the talon?

Reply:
Since you already inspect `data` inside the talon you may as well return `products: data.products || {}` here.